tests/python/gaia-ui-tests/gaiatest/tests/functional/browser/test_browser_bookmark.py

```diff
@@ -17,6 +17,8 @@ def setUp(self):
         GaiaTestCase.setUp(self)
         self.connect_to_local_area_network()
         self.apps.set_permission_by_url(Search.manifest_url, 'geolocation', 'deny')
+        # not sure why the Smart Collection geolocation is being prompted.
+        self.apps.set_permission_by_url(Collection.manifest_url, 'geolocation', 'deny')
```

Comment:
Since the tests are in the gaia repo they are tied together. The gaia master branch has the master tests, the v2.1 gaia branch has the v2.1 tests. 

I would suggest making the test work on master and if its needed on another branch make it compatible at that point.